test/end2end_test.go

```diff
@@ -1333,6 +1333,46 @@ func testConcurrentServerStopAndGoAway(t *testing.T, e env) {
 	awaitNewConnLogOutput()
 }
 
+func (s) TestDetailedConnectionCloseErrorPropagatesToRpcError(t *testing.T) {
+	for _, e := range listTestEnv() {
+		testDetailedConnectionCloseErrorPropagatesToRPCError(t, e)
+	}
+}
+
+func testDetailedConnectionCloseErrorPropagatesToRPCError(t *testing.T, e env) {
```

Comment:
Would you mind rewriting this to use the `StubServer` instead?  See `TestGRPCMethod` for an example.  The boilerplate should be much smaller, and there's no need to test tons of different environments, etc.  You can use `StubServer.S.Stop()` to stop the grpc server abruptly.

Reply:
done